src/_gettsim/ids.py

```diff
@@ -22,27 +20,20 @@ def ehe_id(
     familie__p_id_ehepartner: np.ndarray,
 ) -> np.ndarray:
     """Couples that are either married or in a civil union."""
-    p_id_to_ehe_id: dict[int, int] = {}
-    next_ehe_id = 0
-    result: list[int] = []
-
-    for index, current_p_id in enumerate(map(int, p_id)):
-        current_p_id_ehepartner = int(familie__p_id_ehepartner[index])
-
-        if current_p_id_ehepartner >= 0 and current_p_id_ehepartner in p_id_to_ehe_id:
-            result.append(p_id_to_ehe_id[current_p_id_ehepartner])
-            continue
+    n = np.max(p_id) + 1
+    familie__p_id_ehepartner = np.where(
```

Comment:
```suggestion
    p_id_ehepartner_or_own_p_id = np.where(
```
This is what happens, right? Also, please don't re-use function input names for other things, makes it hard to read (for me).